src/Paket.Core/PublicAPI.fs

```diff
@@ -752,6 +736,42 @@ type Dependencies(dependenciesFileName: string) =
             use fileStream = File.Open (nuspecFile, FileMode.Create)
             doc.Save fileStream
 
+    static member FixNuspecs (referencesFile:ReferencesFile, nuspecFileList:string list) =
+
+        for nuspecFile in nuspecFileList do
+            if not (File.Exists nuspecFile) then
+                failwithf "Specified file '%s' does not exist." nuspecFile
+
+        let directDeps =
+            referencesFile.Groups
+            |> Seq.collect (fun kv -> kv.Value.NugetPackages)
+            |> Seq.map (fun i -> i.Name)
+            |> Set.ofSeq
```

Comment:
This might be a slight change in behavior. I just look into the references file and assume all things there are direct deps (which might not be true but I planned to add a warning when transitives are listed in the references file - should be issue open somewhere)